docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -106,9 +105,9 @@ final outcome depends on how index records are stored in existing index files.
 Two or more index files can be merged with each other, if and only if they all have index records which belong to the
 same bucket (according to index configuration).
 When running `optimizeIndex` command, Hyperspace tries to find such groups of index files and merge them together.
-However, if most of the index files have records from different buckets, then final physical layout of index files
-after running optimize would be similar to the original layout. An example of such an index is an index right after
-creation or full refresh, or an index which is refreshed incrementally with very few additional small source data files.    
+However, if for each given bucket, all index records are already stored in a single index file (i.e. we dont have several index
```

Comment:
` (i.e. it does not have several` ?

Reply:
How about `already stored in a single index file` => `already stored in a single index file for each bucket`??

Reply:
I revised that part again. Plz take a look at it.